### PR TITLE
Random paging

### PR DESCRIPTION
**Pull request by elliott-beach**
This implements random paging, though I don't think it is complete - I don't fully understand the problem yet.

TODO:
[ ] When do we handle loading data from the frame to the page?
[ ] Manage read/write access to pages.

### CODE REVIEW

**Comment by ghost** [comment]
Couple questions:
1. Presumably we only want to evict a page if physical memory is full. Is there an efficient way to keep track of this? i.e. how can we tell which frames are free? Maybe for the first nframes faults we could just assign frames sequentially, before we randomly start evicting them.

2. When do we move data between the disk and physical memory? To do it from the fault handler, which is what I think the intention is,  we would need to make the disk pointer global.

**Comment by ghost** [comment]
For managing read/write access, I think we can just check the permission bits of the page that caused the fault. If its 0 (i.e. it's not in memory), we call set_entry with R, and if it's already R (i.e. already in memory), we call set_entry with R|W.

**Comment by elliott-beach** [comment]
Also I worked on this some more, will push tomorrow. Thanks for checking in.

**Comment by elliott-beach** [comment]
Or we can just dump in everything as R|W, like currently. Do they say that we can't do that?

**Comment by elliott-beach** [comment]
I think this is good to go now.

**Comment by ghost** [comment]
You were right. I accidentally had `page` instead of `curr_page`.

File: main.c [hunk 1]
```diff
@@ -14,15 +14,58 @@ how to use the page table and disk interfaces.
...
+    }
+
+    loadFrameIntoPage(pt, page, frame);
 }
 
-int main( int argc, char *argv[] )
-{
+void loadFrameIntoPage(struct page_table *pt, int page, int frame) {
+    printf("page fault: setting page %d to frame %d\n", page, frame);
+    disk_read(disk, page, page_table_get_physmem(pt) + frame * BLOCK_SIZE);
```
Comment (ghost) [inline main.c:57]: Wouldn't it be more accurate to call this "loadPageIntoFrame"?

**Comment by ghost** [comment]
This looks really good

**Pull request merged**